### PR TITLE
U/tonychoi/v3 cherry pick

### PR DESCRIPTION
**Pull request by Tonewall**
### Issue describing the changes in this PR

resolves #issue_for_this_pr

### Pull request checklist

* [ ] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [ ] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [ ] My changes **do not** require diagnostic events changes
    * Otherwise: I have added/updated all related diagnostic events and their documentation (Documentation issue linked to PR)
* [ ] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

**Comment by yojagad** [comment]
>                     return Encoding.UTF8.GetString(ms.ToArray());

Does this not need to be changed to

`var input = ms.ToArray();
                    if (signature != null && !signature.SequenceEqual(ComputeHMACSHA256(encryptionKey, input)))
                    {
                        throw new InvalidOperationException("Signature mismatches!");
                    }

                    return Encoding.UTF8.GetString(input);`

? #Resolved

---
Refers to: src/WebJobs.Script.WebHost/Security/SimpleWebTokenHelper.cs:87 in 81d6ae4. [](commit_id = 81d6ae4338c32ece9fc2beab2a50be10d7fc036e, deletion_comment = False)

File: src/WebJobs.Script.WebHost/Security/Authentication/Jwt/ScriptJwtBearerExtensions.cs [hunk 1]
```diff
@@ -26,51 +26,63 @@ public static class ScriptJwtBearerExtensions
...
+                };
+            }
 
-        private static TokenValidationParameters CreateTokenValidationParameters()
+            return new string[]
+            {
+                string.Format(SiteAzureFunctionsUriFormat, ScriptSettingsManager.Instance.GetSetting(AzureWebsiteName)),
+                string.Format(SiteUriFormat, ScriptSettingsManager.Instance.GetSetting(AzureWebsiteName))
+            };
+        }
```
Comment (yojagad) [inline src/WebJobs.Script.WebHost/Security/Authentication/Jwt/ScriptJwtBearerExtensions.cs:83]: This look different too

` private static string[] GetValidAudiences()
        {
            if (SystemEnvironment.Instance.IsPlaceholderModeEnabled())
            {
                if (SystemEnvironment.Instance.IsFlexConsumptionSku())
                {
                    return new string[]
                    {
                        ScriptSettingsManager.Instance.GetSetting(WebsitePodName)
                    };
                }
                else if (SystemEnvironment.Instance.IsLinuxConsumptionOnAtlas())
                {
                    return new string[]
                    {
                        ScriptSettingsManager.Instance.GetSetting(ContainerName)
                    };
                }
            }
`

**Comment by Tonewall** [comment]
> > ```
> >                 return Encoding.UTF8.GetString(ms.ToArray());
> > ```
> 
> Does this not need to be changed to
> 
> `var input = ms.ToArray(); if (signature != null && !signature.SequenceEqual(ComputeHMACSHA256(encryptionKey, input))) { throw new InvalidOperationException("Signature mismatches!"); }
> 
> ```
>                 return Encoding.UTF8.GetString(input);`
> ```
> 
> ?
> 
> Refers to: src/WebJobs.Script.WebHost/Security/SimpleWebTokenHelper.cs:87 in [81d6ae4](https://github.com/Azure/azure-functions-host/commit/81d6ae4338c32ece9fc2beab2a50be10d7fc036e). [](commit_id = [81d6ae4](https://github.com/Azure/azure-functions-host/commit/81d6ae4338c32ece9fc2beab2a50be10d7fc036e), deletion_comment = False)

Ah yes, good catch, fixed it! #Resolved

**Review by yojagad — APPROVED** [review]
:shipit:

File: src/WebJobs.Script.WebHost/Security/Authentication/Jwt/ScriptJwtBearerExtensions.cs [hunk 2]
```diff
@@ -26,51 +26,63 @@ public static class ScriptJwtBearerExtensions
...
-                                OnMessageReceived = c =>
-                                {
-                                    // By default, tokens are passed via the standard Authorization Bearer header. However we also support
-                                    // passing tokens via the x-ms-site-token header.
-                                    if (c.Request.Headers.TryGetValue(ScriptConstants.SiteTokenHeaderName, out StringValues values))
-                                    {
-                                        // the token we set here will be the one used - Authorization header won't be checked.
-                                        c.Token = values.FirstOrDefault();
-                                    }
-
```
Comment (mathewc) [inline src/WebJobs.Script.WebHost/Security/Authentication/Jwt/ScriptJwtBearerExtensions.cs:40]: Looks like you're making a bunch of whitespace formatting changes here that should be reverted. Looks like you're removing code formatting?

Comment (mathewc) [reply src/WebJobs.Script.WebHost/Security/Authentication/Jwt/ScriptJwtBearerExtensions.cs:40]: I see now you have an open v3 branch PR https://github.com/Azure/azure-functions-host/pull/9775 open for this. Generally you want to first get that signed off on and in, then pick that signed off change into the hotfix branch.

**Pull request merged**